s3transfer/download.py

```diff
@@ -131,6 +131,27 @@ def _get_temp_fileobj(self):
         return f
 
 
+class DownloadSeekableOutputManager(DownloadOutputManager):
+    @classmethod
+    def is_compatible(self, download_target):
+        return (
+            hasattr(download_target, 'seek') and
+            hasattr(download_target, 'tell')
```

Comment:
Looks like it just tries lseek (seek). https://github.com/python/cpython/blob/master/Modules/_io/fileio.c#L609-L625